cmake/modules/StandaloneCrt.cmake

```diff
@@ -40,129 +40,154 @@ if(MSVC)
-          tvm_micro_add_copy_file(host_isolated_build_deps ${job_src_base}/${copy_src} ${dest_path})
-        endforeach()
-      endforeach()
-    endforeach()
+    set(CRT_LIBRARY_NAME host_standalone_crt_${CRT_LIBRARY})
+    set(CRT_LIBRARY_SOURCES "")
 
-    add_custom_target(standalone_crt DEPENDS ${host_isolated_build_deps})
+    foreach(FILE_NAME IN LISTS ARGN)
+       list(APPEND CRT_LIBRARY_SOURCES ${FILE_NAME})
+    endforeach()
 
-    get_filename_component(host_build_dir_abspath "${CMAKE_CURRENT_BINARY_DIR}/host_standalone_crt" ABSOLUTE)
+    add_library(${CRT_LIBRARY_NAME}
+                STATIC
+                ${CRT_LIBRARY_SOURCES})
 
-    if(${VERBOSE})
-    set(make_quiet QUIET=)
-    else(${VERBOSE})
-    set(make_quiet )
-    endif(${VERBOSE})
+    # add this library to the list of CRT libraries
+    set(CRT_LIBRARIES ${CRT_LIBRARIES} ${CRT_LIBRARY_NAME} PARENT_SCOPE)
 
-    list(APPEND crt_libraries memory graph_executor microtvm_rpc_server microtvm_rpc_common common)  # NOTE: listed in link order.
-    foreach(crt_lib_name IN LISTS crt_libraries)
-      list(APPEND crt_library_paths "host_standalone_crt/lib${crt_lib_name}.a")
-    endforeach()
+    target_include_directories(${CRT_LIBRARY_NAME}
+                               PUBLIC
+                               ${CMAKE_CURRENT_SOURCE_DIR}/src/runtime/micro/
+                               ${STANDALONE_CRT_BASE}/include)
 
-    set(make_common_args
-        "CRT_CONFIG=${CMAKE_CURRENT_SOURCE_DIR}/src/runtime/micro/crt_config.h"
-        "BUILD_DIR=${host_build_dir_abspath}"
-        "EXTRA_CFLAGS=-fPIC"
-        "EXTRA_CXXFLAGS=-fPIC"
-        "EXTRA_LDFLAGS=-fPIC"
-        "${make_quiet}")
-
-    add_custom_command(
-          OUTPUT ${crt_library_paths}
-          COMMAND make ARGS ${make_common_args} clean
-          COMMAND make ARGS ${make_common_args} all
-          WORKING_DIRECTORY "${standalone_crt_base}"
-          DEPENDS standalone_crt ${host_isolated_build_deps})
-
-    add_custom_target(host_standalone_crt DEPENDS ${crt_library_paths})
-
-    foreach(crt_lib IN LISTS crt_libraries)
-      set(cmake_crt_lib_name host_standalone_crt_${crt_lib})
-      list(APPEND cmake_crt_libraries ${cmake_crt_lib_name})
-      add_library(${cmake_crt_lib_name} STATIC IMPORTED GLOBAL)
-      set(cmake_crt_lib_path "${CMAKE_CURRENT_BINARY_DIR}/host_standalone_crt/lib${crt_lib}.a")
-      add_dependencies(${cmake_crt_lib_name} host_standalone_crt "${cmake_crt_lib_path}")
-      set_target_properties(${cmake_crt_lib_name} PROPERTIES
-          IMPORTED_LOCATION "${cmake_crt_lib_path}"
-          IMPORTED_OBJECTS "${cmake_crt_lib_path}"
-          PUBLIC_HEADER "${crt_headers}")
-    endforeach()
+    set_target_properties(${CRT_LIBRARY_NAME}
+                          PROPERTIES
+                          ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/host_standalone_crt
+                          POSITION_INDEPENDENT_CODE ON)
```

Comment:
add option to disable PIC if using a standalone CMake build

Reply:
will do this in subsequent PR that also replaced the standalone_crt/Makefile with cmake version